### PR TITLE
py-pytest-html recipe

### PR DESCRIPTION
**Pull request by BerengerBerthoul**
Recipe for the pure Python pytest-html package

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-pytest-html/package.py [hunk 1]
```diff
@@ -0,0 +1,14 @@
+from spack import *
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-pytest-html/package.py]: Missing a license. You can copy the text from any other file in Spack.

File: var/spack/repos/builtin/packages/py-pytest-html/package.py [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+class PyPytestHtml(PythonPackage):
+    """pytest-html is a plugin for pytest that generates a HTML report for test results"""
+
+    homepage = "https://github.com/pytest-dev/pytest-html"
+    pypi = "pytest-html/pytest-html-3.1.1.tar.gz"
+
+    version('3.1.1', sha256='3ee1cf319c913d19fe53aeb0bc400e7b0bc2dbeb477553733db1dad12eb75ee3')
+
+    depends_on('python@3.6:', type=('build', 'run'))
+    depends_on('py-setuptools', type='build')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-pytest-html/package.py:20]: Missing deps:
```python
    setup_requires=["setuptools_scm"],                                                   
    install_requires=["pytest>=5.0,!=6.0.0", "pytest-metadata"],
```

**Review by adamjstewart — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/py-pytest-html/package.py [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    depends_on('py-setuptools-scm', type='build')
+    depends_on('py-pytest@5.0:5.9,6.1:', type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-pytest-html/package.py]:
```suggestion
    depends_on('py-pytest@5.0:5.999,6.0.1:', type=('build', 'run'))
```

File: var/spack/repos/builtin/packages/py-pytest-metadata/package.py [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+
+class PyPytestMetadata(PythonPackage):
+    """pytest plugin for test session metadata"""
+
+    homepage = "https://github.com/pytest-dev/pytest-metadata"
+    pypi = "pytest-metadata/pytest-metadata-1.11.0.tar.gz"
+
+    version('1.11.0', sha256='71b506d49d34e539cc3cfdb7ce2c5f072bea5c953320002c95968e0238f8ecf1')
+
+    depends_on('python@3.6:', type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-pytest-metadata/package.py]:
```suggestion
    depends_on('python@2.7:2.8,3.6:', type=('build', 'run'))
```

File: var/spack/repos/builtin/packages/py-pytest-metadata/package.py [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+    depends_on('py-setuptools', type='build')
+    depends_on('py-setuptools-scm', type='build')
+    depends_on('py-pytest@2.9:', type=('build', 'run'))
+
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-pytest-metadata/package.py]: This blank line at the end of the file will cause flake8 to fail

**Review by adamjstewart — CHANGES_REQUESTED** [review]

**Review by adamjstewart — DISMISSED** [review]

**Comment by BerengerBerthoul** [comment]
Thanks for the quick feed

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**